lib/src/layout_element.dart

```diff
@@ -31,6 +31,11 @@ class TableLayoutElement extends LayoutElement {
 
   @override
   Widget toWidget(RenderContext context) {
+    final withinAnotherTable = element?.parent?.localName == 'td';
```

Comment:
Hmm it looks like you only look at the direct parent or it's parent parent, instead of recursively going up the tree. Or am I reading this wrong?

I think the approach is _okay_: see if we are already inside another table and if so then fall back to have no size constraints info (even though that can still break if you use this in a horizontal scroll container).